ratelimits/bucket.go

```diff
@@ -320,19 +315,45 @@ func (builder *TransactionBuilder) CertificatesPerDomainTransactions(regId int64
 			}
 			txns = append(txns, txn)
 
-			// Add a spend-only transaction for each per domain bucket.
-			txn, err = newSpendOnlyTransaction(perDomainLimit, perDomainBucketKey, 1)
+			perDomainLimit, err := builder.getLimit(CertificatesPerDomain, perDomainBucketKey)
 			if err != nil {
-				return nil, err
+				if errors.Is(err, errLimitDisabled) {
+					return nil, err
+				}
+				// Add an allow-only transaction for each per domain bucket.
+				txn, err = newAllowOnlyTransaction()
+				if err != nil {
+					return nil, err
+				}
+				txns = append(txns, txn)
```

Comment:
I read this as: "If the limit is disabled, bail out. If we ran into any other error, add a no-op to the list of transactions."

I think the first conditional needs to be inverted -- we should be bailing out on any other error, and only continuing if we got the well-known errLimitDisabled.

Also, I don't think it's necessary to append a no-op transaction to the list of transactions. Given that the "allow-only" transaction doesn't even contain a bucketKey, we can't be using it for better error messages or anything like that. So wouldn't it be exactly the same to not append anything at all?

Reply:
Ah, yes, I've got the conditional wrong here for sure. You also raise a great point, we may be able to eliminate these transactions altogether.